### PR TITLE
Match type

### PR DESCRIPTION
**Pull request by eed3si9n**
Fixes https://github.com/tree-sitter/tree-sitter-scala/issues/205

This implements support for Scala 3 match types.

### CODE REVIEW

File: test/highlight/scala3.scala [hunk 1]
```diff
@@ -23,6 +23,14 @@ object O1:
   //^keyword.function
   //  ^method
 
+object O2:
+  type Elem[A] = A match
+  //^keyword
+  //   ^type.definition
+    case String   => Char
```
Comment (keynmol) [inline test/highlight/scala3.scala:30]: Do the actual types (both on left and right of =>) have highlighting on them? They should, so may be it's worth adding them to the test explicitly.

File: grammar.js [hunk 2]
```diff
@@ -790,8 +804,24 @@ module.exports = grammar({
       field('selector', $._type_identifier),
     ),
 
-    function_type: $ => prec.right(seq(
+    match_type: $ => prec.left(PREC.control, seq(
```
Comment (susliko) [inline grammar.js:807]: Is numeric precedence necessary here? I think, we should minimize its usage as it has a very opaque effect on the result of parsing. 
Maybe, relative precedences (opposed to global) would be a better option

Comment (eed3si9n) [reply grammar.js:807]: Good call. At some point this was conflicting with other types, but it seems like I can remove it now.

Comment (eed3si9n) [reply grammar.js]: Addressed in 41dcb78.

**Review by ckipp01 — APPROVED** [review]
Nice! LGTM!

**Pull request merged**